File: py/IEC61968/Common/Location.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
# local imports
from IEC61968.Common.ConfigurationEvent import ConfigurationEvent
from IEC61968.Common.CoordinateSystem import CoordinateSystem
from IEC61968.Common.Crew import Crew
from IEC61968.Common.ElectronicAddress import ElectronicAddress
from IEC61968.Common.PositionPoint import PositionPoint
from IEC61968.Common.Status import Status
from IEC61968.Common.StreetAddress import StreetAddress
from IEC61968.Common.TelephoneNumber import TelephoneNumber
from IEC61970.Base.Meas.Measurement import Measurement
from IEC61970.Base.Core.IdentifiedObject import IdentifiedObject
from IEC61970.Base.Core.PowerSystemResource import PowerSystemResource
from IEC61968.Operations.SwitchingOrder import SwitchingOrder


class Location(IdentifiedObject):

    def __init__(self):
        super().__init__()
        self.direction = ""
        self.electronic_address = ElectronicAddress()
        self.geo_info_reference = ""
        self.main_address = StreetAddress()
        self.phone1 = TelephoneNumber()
        self.phone2 = TelephoneNumber()
        self.secondary_address = StreetAddress()
        self.status = Status()
        self.type = ""
        self.measurements = Measurement()
        self.coordinate_system = CoordinateSystem()
        self.power_system_resources = PowerSystemResource()
        self.configuration_events = ConfigurationEvent()
        self.crew = Crew()
        self.switching_order = SwitchingOrder()
        self.position_points = PositionPoint()

File: java/IEC61968/Common/Location.java
package IEC61968.Common;

import IEC61970.Base.Domain.String;
import IEC61970.Base.Meas.Measurement;
import IEC61970.Base.Core.IdentifiedObject;
import IEC61970.Base.Core.PowerSystemResource;
import IEC61968.Operations.SwitchingOrder;

/**
 * The place, scene, or point of something where someone or something has been, is,
 * and/or will be at a given moment in time. It can be defined with one or more
 * postition points (coordinates) in a given coordinate system.
 * @author T. Kostic
 * @version 1.0
 * @created 25-Dec-2023 8:45:22 PM
 */
public class Location extends IdentifiedObject {

	/**
	 * (if applicable) Direction that allows field crews to quickly find a given asset.
	 * For a given location, such as a street address, this is the relative direction
	 * in which to find the asset. For example, a streetlight may be located at the
	 * 'NW' (northwest) corner of the customer's site, or a usage point may be located
	 * on the second floor of an apartment building.
	 */
	public String direction;
	/**
	 * Electronic address.
	 */
	public ElectronicAddress electronicAddress;
	/**
	 * (if applicable) Reference to geographical information source, often external to
	 * the utility.
	 */
	public String geoInfoReference;
	/**
	 * Main address of the location.
	 */
	public StreetAddress mainAddress;
	/**
	 * Phone number.
	 */
	public TelephoneNumber phone1;
	/**
	 * Additional phone number.
	 */
	public TelephoneNumber phone2;
	/**
	 * Secondary address of the location. For example, PO Box address may have
	 * different ZIP code than that in the 'mainAddress'.
	 */
	public StreetAddress secondaryAddress;
	/**
	 * Status of this location.
	 */
	public Status status;
	/**
	 * Classification by utility's corporate standards and practices, relative to the
	 * location itself (e.g., geographical, functional accounting, etc., not a given
	 * property that happens to exist at that location).
	 */
	public String type;
	public Measurement Measurements;
	/**
	 * Coordinate system used to describe position points of this location.
	 */
	public CoordinateSystem CoordinateSystem;
	/**
	 * All power system resources at this location.
	 */
	public PowerSystemResource PowerSystemResources;
	/**
	 * All configuration events created for this location.
	 */
	public ConfigurationEvent ConfigurationEvents;
	public Crew Crew;
	public SwitchingOrder SwitchingOrder;
	/**
	 * Sequence of position points describing this location, expressed in coordinate
	 * system 'Location.CoordinateSystem'.
	 */
	public PositionPoint PositionPoints;

	public Location(){

	}
}//end Location